tests/testsuite/shell_quoting.rs

```diff
@@ -26,14 +25,24 @@ fn features_are_quoted() {
         .build();
 
     p.cargo("check -v")
-            .env("MSYSTEM", "1")
-            .with_status(101)
-            .with_stderr_contains(
-                r#"[RUNNING] `rustc [..] --cfg 'feature="default"' --cfg 'feature="some_feature"' [..]`"#
-            ).with_stderr_contains(
-                r#"
+        .env("MSYSTEM", "1")
+        .with_status(101)
+        .with_stderr_data(str![])
+        .with_stderr_data(str![[r#"
+[CHECKING] foo v0.1.0 ([ROOT]/foo)
+[RUNNING] `rustc --crate-name foo --edition=2015 src/main.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type bin --emit=dep-info,metadata -C embed-bitcode=no -C debuginfo=2 --cfg 'feature="default"' --cfg 'feature="some_feature"' -C metadata=c2925e00c1458fcb -C extra-filename=-c2925e00c1458fcb --out-dir [ROOT]/foo/target/debug/deps -L dependency=[ROOT]/foo/target/debug/deps`
```

Comment:
We don't have redactions for hashes (metadata and extra-filename) for now. Let's redact most of those args as before?

```suggestion
[RUNNING] `rustc [..] --cfg 'feature="default"' --cfg 'feature="some_feature"' [..]`
```

Reply:
Should we work to address to auto-redact these first?  The downside is delaying @henry40408's contribution.  The upside is we don't have to remember to update these later.

Hmm, maybe this isn't a problem.  If we add auto-redactions later, this will still work.  If/when the output changes, the snapshot updating will take care of it for us.  The only benefit remaining is helping with future test ports and so auto-redacting now is at a "nit" level and I don't see it as a blocker.

Reply:
I am fine with moving this forward now and come back later